Samples/src/UnitTesting/CCNOTCircuits.qs

```diff
@@ -331,30 +261,23 @@ namespace Microsoft.Quantum.Samples.UnitTesting {
     /// #Recall that ExpFrac used in the circuit below is an exponent of the
     ///  respective multi-qubit Pauli gate times numerator Pi() i /2^{n-1}
     ///  It is a primitive gate implemented by Quantum.Intrinsics
-    operation CCZ1 (qubit1 : Qubit, qubit2 : Qubit, qubit3 : Qubit) : Unit {
-
-        body (...) {
-            let register = [qubit1, qubit2, qubit3];
-
-            // note that CCZ = exp( iπ|1⟩⟨1|⊗|1⟩⟨1|⊗|1⟩⟨1| )
-            // next use |1⟩⟨1| = (I-Z)/2 and write
-            // iπ|1⟩⟨1|⊗|1⟩⟨1|⊗|1⟩⟨1| =
-            //          = iπ/2³(I⊗I⊗I - Z⊗I⊗I - I⊗Z⊗I - I⊗I⊗Z + Z⊗Z⊗I + Z⊗I⊗Z + I⊗Z⊗Z - Z⊗Z⊗Z)
-            // using above we express CCZ as:
-            // exp( iπ/2³I⊗I⊗I )
-            ExpFrac([PauliI, PauliI, PauliI], 1, 3, register);
-            ExpFrac([PauliZ, PauliI, PauliI], -1, 3, register);
-            ExpFrac([PauliI, PauliZ, PauliI], -1, 3, register);
-            ExpFrac([PauliI, PauliI, PauliZ], -1, 3, register);
-            ExpFrac([PauliZ, PauliZ, PauliI], 1, 3, register);
-            ExpFrac([PauliZ, PauliI, PauliZ], 1, 3, register);
-            ExpFrac([PauliI, PauliZ, PauliZ], 1, 3, register);
-            ExpFrac([PauliZ, PauliZ, PauliZ], -1, 3, register);
-        }
-
-        adjoint self;
```

Comment:
self should be preserved for optimization purposes (it's tracked as inferred property in the compiler)